tests/api_connexion/endpoints/test_connection_endpoint.py

```diff
@@ -200,14 +229,215 @@ def _create_connections(self, count):
 
 
 class TestPatchConnection(TestConnectionEndpoint):
-    @unittest.skip("Not implemented yet")
-    def test_should_response_200(self):
-        response = self.client.patch("/api/v1/connections/1")
+
+    @provide_session
+    def test_patch_should_response_200(self, session):
+        self._create_connection(session)
+        payload = {
+            "connection_id": "test-connection-id",
+            "conn_type": 'test_type',
+            "extra": "{'key': 'var'}"
+        }
+        response = self.client.patch("/api/v1/connections/test-connection-id",
+                                     json=payload)
+        assert response.status_code == 200
+
+    @provide_session
+    def test_patch_should_response_200_with_update_mask(self, session):
+        self._create_connection(session)
+        test_connection = "test-connection-id"
+        payload = {
+            "connection_id": test_connection,
+            "conn_type": 'test_type_2',
+            "extra": "{'key': 'var'}",
+            'login': "login",
+            "port": 80,
+        }
+        response = self.client.patch(
+            "/api/v1/connections/test-connection-id?update_mask=port,login",
+            json=payload
+        )
         assert response.status_code == 200
+        connection = session.query(Connection).filter_by(conn_id=test_connection).first()
+        self.assertEqual(connection.password, None)
+        self.assertEqual(
+            response.json,
+            {
+                "connection_id": test_connection,  # not updated
+                "conn_type": 'test_type',  # Not updated
+                "extra": None,  # Not updated
+                'login': "login",  # updated
+                "port": 80,  # updated
+                "schema": None,
+                "host": None
+
+            }
+        )
+
+    @parameterized.expand(
+        [
+            (
+                {
+                    "connection_id": 'test-connection-id',
```

Comment:
Can you add a test that checks the request without the connection_id key?

Reply:
Fixed.https://github.com/apache/airflow/pull/9266/commits/1a4efd35d37bae28735af470d7846ff3b0bce6f6.
It happened that the required fields I added earlier in spec was affecting the update. I have removed it. It would have been fine in the spec to have the required field if there's a separate schema for updating. But with manual validation, everything seems fine. Please check